### ISSUE TITLE
Mobile App: Matchmaking (Studies) Iteration 2

### ISSUE THREAD

**Issue by samgawaran** [colonistio/design#272]
## Goals
- Newbie and Pro players should understand the flow 
- Give players enough information that they are matchmaking
- Matchmaking should be seamless on a homepage orientation we've finalized
- Players can Cancel or Go Back if they miss-pressed a game mode

## Needs
- Consistency with the brand
- Propose solutions that use our current assets as much as you can

## Principles
- [Hierarchy](https://www.uxdesigninstitute.com/blog/ux-design-principles/)
- [Contrast](https://www.toptal.com/designers/gui/principles-of-design-infographic#:~:text=There%20are%20twelve%20basic%20principles,that%20make%20sense%20to%20users.)
TLDR: various differences in the design makes them stand out more
- [Efficiency](https://www.uxdesigninstitute.com/blog/ux-design-principles/)
TLDR: Does the design of the product allow the user to complete their desired tasks quickly and efficiently?
- [Visibility of system status](https://www.nngroup.com/articles/ten-usability-heuristics/)
- [Movement](https://www.toptal.com/designers/gui/principles-of-design-infographic#:~:text=There%20are%20twelve%20basic%20principles,that%20make%20sense%20to%20users.)

## Research
![image](https://user-images.githubusercontent.com/90673012/213644208-a0cf09d7-77a6-4998-b1c9-5a169fba54ee.png)

- Shows an `estimated time` or `time elapsed` after you pressed the Matchmaking button
- Overwatch (Desktop) and Extreme Tennis (iOS) allow players to go to different pages while matchmaking
- All of them allows player to cancel matchmaking as long as there's no players yet
- Magnifying glass/search icon is the most commonly used icon for matchmaking

## Solutions
- Further optimization would be needed after we finalize the homepage to ensure that the experience is smooth and seamless.
- this submission lists out possible look where our matchmaking indicator is not shown in our usual popup

### Solution 1: Full Screen - Show Avatars
![image](https://user-images.githubusercontent.com/90673012/213633844-b32cd91a-4493-41af-b5aa-dc6d48a0034b.png)
![image](https://user-images.githubusercontent.com/90673012/213638276-c7486daf-a4d5-45f9-bc8c-049074c35d44.png)

- Players will see the player avatar of their opponents.
- for Ranked mode, will show division icon and rating '
- Players can't cancel once someone is matchmaked with them

Followed Principles: Hierarchy, Movement, Contrast
Ignored Principles: 

### Solution 2: Full Screen: Show Elapsed Time
![image](https://user-images.githubusercontent.com/90673012/213638388-1e25e345-977b-45dd-be57-1871daecbe05.png)

- Easier to implement compared to solution 1
- Shows elapsed time could possibly be better than showing avatars. atleast with showing elapsed time, the player knows that the matchmaking is still ongoing.

Followed Principles: Visibility of System Status, Movement, Contrast
Ignored Principles: 

### Solution 3: Collapsible Widget
![image](https://user-images.githubusercontent.com/90673012/213639774-0b49adc7-fad5-460d-8d38-0048b82afb14.png)

- Games like Overwatch, Extreme Tennis and Mobile Legends allow players to continue browsing other pages after they matchmake. 
- Useful for game modes with long waiting time (ex. Ranked C&K where I know some players that mentioned that they waited for about 5 minutes to just to match up)
- Color should not blend too much, the purpose of this solution is to let the player browse, not for them to constantly notice the widget
- Easiest in terms of how close it is in the current design, this way, the experience between Desktop and Mobile matchmaking wouldn't be too different

Followed Principles: Movement, Efficiency
Ignored Principles: Visibility of System Status, Contrast

## Conclusion
Proposed Solution: `Solution 2: Showing Elapse Time`

Benefits:
- not interactive as `solutions 1` but will be much easier to implement for an MVP
- fewer edge cases since it's full screen. We won't worry too much about popups and such
- having a timer is proven to increase game retention: https://www.gamedeveloper.com/design/time-for-a-timer---effective-use-of-timers-in-game-design

Potential Risks: 
- New design. We need to adjust the Desktop version or else the UX would be too different

Figma link: 
- https://www.figma.com/file/4sPdkWQrqoiGgFkMLtVSqr/Colonist-UI%2FUX-App?node-id=647%3A96480&t=cDz1UBh738nmDEAW-1

Loom Video explaining reasons for the design:
https://www.loom.com/share/90b9295d2fd2449aa22ede020b857d1f

## Considered
- [x] This proposal fits [company vision](https://github.com/colonistio/katan/blob/development/docs/company/company-values.md)
- [x] This proposal is impactful according to [Prioritizer Sheet](https://docs.google.com/spreadsheets/d/1A2qZ-KG4E2ma8TdnP_9LayMfnEfLVhmoM5_SgazXYHw/edit#gid=0)
- [x] This submission abides by [Des-lvl-1](https://github.com/colonistio/katan/blob/development/docs/process/designer-level-1.md), [Des-lvl-2](https://github.com/colonistio/katan/blob/development/docs/process/designer-level-2.md) & [Des-lvl-3](https://github.com/colonistio/katan/blob/development/docs/process/designer-level-3.md)

## Reviewers
@demiculus @JuanRoman77 @woodifred @5econds

**Comment by JuanRoman77** [comment]
- Why show elapsed time? That could make people anxious. Have an hourglass or some other kind of animated icon
- Definitely not showing avatars, nor disabled the cancel button
- Cancel button shouldn't be green

**Comment by demiculus** [comment]
The goal for matchmaking screen is to `get players inside the game`. That is our one and only goal.
We don't want the player exiting, leaving the app etc.. our only goal is to get players inside the game.

Then the players have needs on that page.
- They want to be able to cancel if necessary

I can't think of anything else they'd need to do.

Every other element is determined depending on if it makes the player more likely to go in or not.

**Comment by demiculus** [comment]
1. Elapsed time -> From research, some of the games show some of them don't show. 
    - No timer makes you feel abandoned
    - most games have time related either elapsed or estimated
    - we can abtest it later
    - Go with elapsed time
2. Drawer vs fullscreen -> Good choice with fullscreen

> Definitely not showing avatars, nor disabling the cancel button
Cancel button shouldn't be green

Agree
